Feature/mechanics test

MatiasCodas opened the pull request.

MatiasFCodas committed:
start implementing initial features

MatiasFCodas committed:
Start implementing movement player

Cronokami committed:
Start implementing animation

Also tried to get used with the new code patterns

Cronokami committed:
Updating testing scene for the new animation

Name is "MainScene"

Cronokami committed:
Implementing basic movement

A few additions into the scene and animation to help with visualizing the movement

MatiasFCodas committed:
start prototyping the mechanics

Cronokami committed:
prototyping rope curse

Cronokami committed:
Multiple ropes

Cronokami committed:
Rope with curses and collision

Cronokami committed:
Optimized rope

and a bit heavier

Cronokami committed:
Renaming scripts for new project(for clarity)

Cronokami committed:
Sending curse message from tip

MatiasCodas commented:
![Captura de tela 2022-11-30 103652](https://user-images.githubusercontent.com/55106956/204810837-d762efa2-e88b-4aa7-a41b-c0fa2cf0834e.png)

https://www.youtube.com/watch?v=_6Jq1wliUVQ

Não seria melhor ser um hinge Joint que nem no video acima?  O spring é uma mola, acho que faz mais sentido ser uma dobradiça

MatiasCodas commented:
![Captura de tela 2022-11-30 104118](https://user-images.githubusercontent.com/55106956/204811476-39a3a788-67b4-4f35-a9f1-2772524e7fb5.png)

Não ta funcionando a maldição quando conecta

Cronokami committed:
Lessening spring elasticity and renaming functions

Cronokami commented:
Sobre as joints, eu tinha colocado como spring pra termos mais liberdade caso a gente faça algo mais elástico, possivelmente um estilingue ou algo do tipo.
Tentei ajustar a spring pra ter menos elasticidade, mas se preferir deixar de hinge pra ter ctz é fácil de mudar.

Cronokami commented:
E a maldição eu tinha desabilitado pra poder testar a cordinha, mas tava pensando em colocar a função da maldição no item como "Curse1()".
Coloquei pra corda mandar um SendMessage quando ela sai de um item amaldiçoado.
Fiz isso pq o Magno tinha falado sobre a maldição fazer coisas diferentes dependendo do item que acerta, tipo ela explodir uma pedra se acertasse nela, em vez de só destruir.

Cronokami committed:
Basic object being affected by curse script

Cronokami committed:
Script added to objects

Cronokami committed:
Basic enemy and stones for rope to interact

Cronokami committed:
Demoday Art implemented and demoday adjusts

Cronokami committed:
Changed rope target order

and fixed bugs in the process

Cronokami committed:
Getting camera farther and implementing new movement and phys on player

Cronokami committed:
Basic new rope implemented

known issues:
some console errors about the curse projectile
rope can drift around targets

Cronokami committed:
Updating rope with minor advances with the physics(bad performance)

Cronokami committed:
Curse and rope getting tangled(buggy af but not as perf intensive)

Cronokami committed:
Start of the "tangle" mechanic (and interactions with the rope)

Also the perfomance is way better now

Cronokami committed:
Tangle finished?

Doesn't seem too performance intensive

Cronokami committed:
Rope going back prototybe

still not working properly, stuff with layers and force directions
next update on this should have both bugs in mind
 - rope being thrown again when calling back
 - rope getting crazy when hitting something new

Cronokami committed:
Rope going back and creating a ScriptObj to save the stats

Cronokami committed:
Rope goes back when an enemy is destroyed

took some time tho, lots of bugs

Cronokami committed:
Code cleanup and bug fixing

Cronokami committed:
Cursing objects previously roped

Cronokami committed:
Creating Spawn System

Very buggy, only one at a time for now but most of it should be ready I think

Cronokami committed:
Enemy Spawn working

And trying to start with the hud but failing

Cronokami committed:
Trying to make the assembly work
